stackslib/src/net/api/postblock_v3.rs

```diff
@@ -0,0 +1,185 @@
+    fn parse_postblock_octets(mut body: &[u8]) -> Result<NakamotoBlock, Error> {
+        let block = NakamotoBlock::consensus_deserialize(&mut body).map_err(|e| {
+            if let CodecError::DeserializeError(msg) = e {
+                Error::DecodeError(format!("Failed to deserialize posted transaction: {}", msg))
+            } else {
+                e.into()
+            }
+        })?;
+        Ok(block)
+    }
+}
+
+/// Decode the HTTP request
+impl HttpRequest for RPCPostBlockRequestHandler {
+    fn verb(&self) -> &'static str {
+        "POST"
+    }
+
+    fn path_regex(&self) -> Regex {
+        Regex::new(&format!("^{PATH}$")).unwrap()
+    }
+
+    fn metrics_identifier(&self) -> &str {
+        PATH
+    }
+
+    /// Try to decode this request.
+    /// There's nothing to load here, so just make sure the request is well-formed.
+    fn try_parse_request(
+        &mut self,
+        preamble: &HttpRequestPreamble,
+        _captures: &Captures,
+        query: Option<&str>,
+        body: &[u8],
+    ) -> Result<HttpRequestContents, Error> {
+        if preamble.get_content_length() == 0 {
+            return Err(Error::DecodeError(
+                "Invalid Http request: expected non-zero-length body for PostBlock".to_string(),
+            ));
+        }
+
+        if preamble.get_content_length() > MAX_PAYLOAD_LEN {
+            return Err(Error::DecodeError(
+                "Invalid Http request: PostBlock body is too big".to_string(),
+            ));
+        }
+
+        if Some(HttpContentType::Bytes) != preamble.content_type || preamble.content_type.is_none()
+        {
+            return Err(Error::DecodeError(
+                "Invalid Http request: PostBlock takes application/octet-stream".to_string(),
+            ));
+        }
+
+        let block = Self::parse_postblock_octets(body)?;
+
+        self.block = Some(block);
+        Ok(HttpRequestContents::new().query_string(query))
+    }
+}
+
+impl RPCRequestHandler for RPCPostBlockRequestHandler {
+    /// Reset internal state
+    fn restart(&mut self) {
+        self.block = None;
+    }
+
+    /// Make the response
+    fn try_handle_request(
+        &mut self,
+        preamble: HttpRequestPreamble,
+        _contents: HttpRequestContents,
+        node: &mut StacksNodeState,
+    ) -> Result<(HttpResponsePreamble, HttpResponseContents), NetError> {
+        // get out the request body
+        let block = self
+            .block
+            .take()
+            .ok_or(NetError::SendError("`block` not set".into()))?;
+
+        let response = node
+            .with_node_state(|network, sortdb, chainstate, _mempool, _rpc_args| {
+                let mut handle_conn = sortdb.index_handle_at_tip();
+                Relayer::process_new_nakamoto_block(
+                    &network.burnchain,
+                    &sortdb,
+                    &mut handle_conn,
+                    chainstate,
+                    &block,
+                    None,
```

Comment:
Should this also wake up the chains coordinator thread?

Reply:
That happens in the net::relay module when the network result is handled (https://github.com/stacks-network/stacks-core/pull/4902/files#diff-a29f0a1f3db4cbe0a08f45ff998bf03fd639c1b8608c362a421639292769a51fR1988)